fe/fe-core/src/main/java/com/starrocks/analysis/MatchPredicate.java

```diff
@@ -0,0 +1,216 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.analysis;
+
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.starrocks.catalog.ArrayType;
+import com.starrocks.catalog.Function;
+import com.starrocks.catalog.FunctionSet;
+import com.starrocks.catalog.ScalarFunction;
+import com.starrocks.catalog.Type;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.thrift.TExprNode;
+import com.starrocks.thrift.TExprNodeType;
+import com.starrocks.thrift.TExprOpcode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Objects;
+
+/**
+ * Fulltext match predicate
+ */
+public class MatchPredicate extends Predicate {
+    private static final Logger LOG = LogManager.getLogger(MatchPredicate.class);
+
+    public enum Operator {
+        MATCH_TERMS("MATCH_TERMS", "match_terms", TExprOpcode.MATCH_TERMS),
+        MATCH_ALL("MATCH_ALL", "match_all", TExprOpcode.MATCH_ALL),
+        MATCH_PHRASE("MATCH_PHRASE", "match_phrase", TExprOpcode.MATCH_PHRASE),
+        MATCH_FUZZY("MATCH_FUZZY", "match_fuzzy", TExprOpcode.MATCH_FUZZY),
```

Comment:
1. Is the `MATCH_FUZZY` same with `like`, `MATCH_WILDCARD` same with `regex_like`?
2. I send a query `select * from test where match_fuzzy(f1, "thinkbad", 1)`. If there is no inverted index of f1, what's the behavior?
3. When to determine use inverted index or not if `f1` already has an inverted index? Depends on the cardinality of `f1` or use the inverted index in any cardinality?
4. `select * from test where match_fuzzy(f1, "thinkbad", 1) and f2 = 'xxx'` If `f1` has inverted index, `f2` has bitmap index, how to determine the filter order?